controllers/machinedeletionremediation_controller.go

```diff
@@ -76,25 +91,43 @@ type MachineDeletionRemediationReconciler struct {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
-func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (finalResult ctrl.Result, finalErr error) {
 	log := r.Log.WithValues("machinedeletionremediation", req.NamespacedName)
 
 	log.Info("reconciling...")
 
 	var err error
 	var remediation *v1alpha1.MachineDeletionRemediation
 	if remediation, err = r.getRemediation(ctx, req); remediation == nil || err != nil {
-		if apiErrors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
 		return ctrl.Result{}, err
 	}
 
 	log.Info("Machine Deletion Remediation CR found", "name", remediation.GetName())
 
+	defer func() {
+		if updateErr := r.updateStatus(ctx, remediation); updateErr != nil {
+			if !apiErrors.IsConflict(updateErr) {
+				finalErr = utilerrors.NewAggregate([]error{updateErr, finalErr})
+			}
+			finalResult.RequeueAfter = time.Second
+		}
+	}()
+
 	// Remediation's name was created from Node's name
 	nodeName := remediation.GetName()
 
+	if r.isStoppedByNHC(remediation) {
+		log.Info("NHC stop requested")
+		r.updateConditions(remediationTerminatedByNHC, remediation)
```

Comment:
at the moment, `updateConditions` returns error only if the `processingChangeReason` is unknown, which is unexpected and there's no way to recover it.

Reply:
Not reviewers only, but also linters might complain about ignoring return values.
I'd suggest to at least add `_, _ = ...`, in order to indicate that it's on purpose.
However, I would prefer to handle the error as well, because today we know what can happen or not, but tomorrow we might change that and forget this place. I would log and return the error.

Reply:
same applies to all other occurrences of updateConditions()

Reply:
reconsidering this case, I think that returning `updateConditions` error (in case of error) is the best option anyway. Right now, the error returned is not recoverable, but still I think it isn't good to continue like nothing happened, and if the function changes in the future, the error handling will be correct already.